### PR TITLE
another round of fixes for Sonar warnings

### PR DESCRIPTION
**Pull request by vladak**
Trying to address more Sonar blockers.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java [hunk 1]
```diff
@@ -140,8 +140,12 @@ Executor getHistoryLogExecutor(final File file, String sinceRevision) throws IOE
         } else {
             argv.add(file.getName());
         }
-        if (sinceRevision != null && new Scanner(sinceRevision).hasNextInt()) {
-            argv.add("-v" + (Integer.parseInt(sinceRevision) + 1) + ":" + Integer.MAX_VALUE);
+        if (sinceRevision != null) {
+            try (Scanner scanner = new Scanner(sinceRevision)) {
+                if (scanner.hasNextInt()) {
+                    argv.add("-v" + (Integer.parseInt(sinceRevision) + 1) + ":" + Integer.MAX_VALUE);
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java:146]: I'd use here Scanner's `nextInt` method:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Scanner.html#nextInt()

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java [hunk 2]
```diff
@@ -249,8 +252,12 @@ boolean fileHasAnnotation(File file) {
                     if (parts[0].equals(file.getName())) {
                         // Check if the version field is greater than 1
                         //  which indicates that annotate will work
-                        if (parts.length > 2 && new Scanner(parts[2]).hasNextInt()) {
-                            return Integer.parseInt(parts[2]) > 1;
+                        if (parts.length > 2) {
+                            try (Scanner scanner = new Scanner(parts[2])) {
+                                if (scanner.hasNextInt()) {
+                                    return Integer.parseInt(parts[2]) > 1;
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java:258]: Again, I'd go with `nextInt`

**Review by ahornace — APPROVED** [review]
Good job with the cleanups! :)

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java (hunk 1 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java:146]: Then I'd have to handle the exceptions which would make the code a bit more difficult to read, I think.

Comment (ahornace) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/SSCMRepository.java:146]: Oh, I see, forgot about the exception, then let's leave it as it is

**Pull request merged**